### PR TITLE
refactor to remove docker and per-fedora package builds

### PR DESCRIPTION
**Pull request by joemiller**
The goal for the wp-cli RPM is to download wp-cli.phar, and then
package it into an RPM. There is no compilation happening and no
linking against fedora libraries, thus there is no reason to use the
complicated docker build method.

### CODE REVIEW

**Comment by pantheon-circleci** [comment]
@joemiller, thanks for your PR! By analyzing the blame information on this pull request, we identified @greg-1-anderson to be a potential reviewer

**Comment by joemiller** [comment]
ping @RazzYoshi @greg-1-anderson This is how we should package drush and wp-cli. We do not need docker nor the complicated machinations that come with it.

The diff is probably unhelpful. View the branch in github instead =)  https://github.com/pantheon-systems/rpmbuild-wp-cli/tree/simplify-no-docker

**Comment by spheromak** [comment]
:+1: :shipit: :rocket: didn't  know this stuff wasn't needing to link or build anything.

**Comment by spheromak** [comment]
ping @joemiller this still something we want to merge ?

**Comment by greg-1-anderson** [comment]
I'm going to start looking at this today. We do want this change; just need to test it to ensure that it works okay.

**Comment by greg-1-anderson** [comment]
I think this is ready to merge now.

**Comment by joemiller** [comment]
@greg-1-anderson looks good to me. +1. I had a note on my TODO list to "fix the rpm name", because I think you said I changed the name of the rpm in the initial refactor. Is that the case?

**Comment by joemiller** [comment]
@greg-1-anderson oh i see. The name is "wp-cli-$CHANNEL". Do we need that channel? I thought we didn't for this one, and it was only a holdover from the copy-pasta of fusedav's rpm script ?

**Comment by greg-1-anderson** [comment]
Hm, the Drush rpm script does NOT put the channel into the name. Let me double-check this...

**Comment by greg-1-anderson** [comment]
I did confirm that the rpm name is unchanged from what it was before; I remember making that comment to you, and it was from memory, so I suspect I was simply mistaken.

I don't think we need $CHANNEL in the name; however, it is in the name in the rpm that is currently installed.  We could change it, but then we'd have to change the [wp-cli cookbook](https://github.com/pantheon-cookbooks/wp-cli/blob/master/recipes/default.rb#L9).  I'm not sure if there would be some extra operations we'd need to do to avoid conflicts between the "new" and the "old" rpm names.

Maybe just leave it as it is?

**Comment by joemiller** [comment]
@greg-1-anderson looking on a prod endpoint, we have these 2 rpm's installed:

```
[joe@endpoint0e7de30d]~% rpm -qa | grep wp-cli
wp-cli-0.x-release-0.23.0-1466808975.61ee56e.x86_64
wp-cli-0.5.0-201512092043.git9f7ff51.x86_64
```

and their contents:

```
[joe@endpoint0e7de30d]~% rpm -ql wp-cli-0.x-release
/opt/pantheon/wp-cli-0.x/wp-cli.phar

## ^^ I actually don't think the channels would have worked as intended. If we tried to install a new rpm with name 'wp-cli-0.x-devel' and it used the same path '/opt/pantheon/wp-cli-0.x/' as the -release rpm I think rpm would exit with error due to conflict of two packages trying to "own" the same file.
```

```
[joe@endpoint0e7de30d]~% rpm -ql wp-cli
/opt/wp-cli/wp-cli.phar
```

I don't think we need the $CHANNEL and I'd prefer to drop them as it drops a lot of code and conditionals and context-dependent builid logic out of the codebase. We should be able to do that by removing the old rpm then installing the new rpm in chef. Chef will execute these in the specified order:

note - this is just pseudocode, i haven't looked at the wp-cli cookbook:

```
# TODO: this remove-resource can be deleted from the cookbook after all nodes have executed this recipe at least once (though it could also be left in forever too =)
package 'wp-cli-0.x-release' do
  action :remove
end

package 'wp-cli-0.x' do
  action :upgrade
end
```

**Comment by greg-1-anderson** [comment]
The wp-cli-0.5.0-blah is left over.  I guess there needs to be an `action :remove` for that one as well.  I'll push another commit here to remove $CHANNEL, then make a PR to fix the cookbook.

**Comment by joemiller** [comment]
@greg-1-anderson you are the best. thank you for pushing this over the finish line.

File: scripts/build-rpm.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+bin="$(cd -P -- "$(dirname -- "$0")" && pwd -P)"
+
+# set a default build -> 0 for when it doesn't exist
+CIRCLE_BUILD_NUM=${CIRCLE_BUILD_NUM:-0}
+
+# epoch to use for -revision
+epoch=$(date +%s)
+
+shortname="wp-cli-0.x"
+name="$shortname"
```
Comment (joemiller) [inline scripts/build-rpm.sh:14]: i think $name is now unused.

**Comment by joemiller** [comment]
👍

File: scripts/build-rpm.sh (hunk 1 above)
Comment (greg-1-anderson) [inline scripts/build-rpm.sh:14]: True; I'm just going to leave them separate for now, though.

**Pull request merged**